### DISCUSSION TITLE
BT PIN Code on RAK4631 Display

### DISCUSSION THREAD

**Discussion by KC9SEB** [markqvist/Reticulum#632] [Ideas]
For those who have a display attached, would be great to have PIN code presented on display at pairing; just like it does with Heltec today.

**Comment by michael23505** [comment]
I have about 5 of the rak 4631's without displays and haven't been able to pair over usb or serial. i think with a display you have to hook up to your phone and go into hardware settings on sideband, and go to rnode, and click enable bluetooth and start pairing. i'm hoping to order a display soon so i can test this out. would love to have a few working rnodes to play with now that gps works.

**Reply by KC9SEB** [reply]
I have a RAK with display - Able to pair successfully - but am relying upon the pin code from the cli output of rnodeconf -p. My original ask was for the pin code to be displayed on the led - same behavior as what I am getting on a Heltec V3 board.

**Comment by markqvist** [comment]
If the device does not have a display, the only way to get the pairing PIN is to use `rnodeconf` to put the device into pairing mode, which will print out the PIN to the console, or to use the display read functionality of @liamcottle's web-flasher.

**Comment by michael23505** [comment]
found updated version of web flasher that has the read display capability. display says firmware corrupt. i think it could be flashing in windows might switch com ports during reboot, and the process can't finish properly. i'm going to try in linux soon.

**Reply by KC9SEB** [reply]
I had the same result initially with the RAK, but was able to see that message on the display (firmware corrupt), was using a MAC if that helps. I had to use Liam's flasher to "provision" and "set firmware hash" steps to resolve it - I have a fully functional node - but would still like the pairing code to be displayed on the node at pairing time if possible.

**Comment by michael23505** [comment]
thanks. provisioning and set firmware hash helped, but i still get a failed to get bluetooth pin timeout message, and i can't see a pin on the display. it does work on an android phone over usb though, so that's progress.

**Comment by michael23505** [comment]
after playing around with it i managed to get bluetooth working on all 5 rak4631's. having some form of display is definitely necessary.
